Mathlib/Algebra/Lie/OfAssociative.lean

```diff
@@ -205,6 +205,40 @@ theorem LieSubalgebra.toEnd_mk (K : LieSubalgebra R L) {x : L} (hx : x ∈ K) :
     LieModule.toEnd R K M ⟨x, hx⟩ = LieModule.toEnd R L M x :=
   rfl
 
+section IsFaithful
+
+open Function
+
+namespace LieModule
+
+/-- A Lie module is *faithful* if the associated map `L → End M` is injective. -/
+@[mk_iff]
+class IsFaithful : Prop where
+  injective_toEnd : Injective <| toEnd R L M
+
+@[simp]
+lemma toEnd_eq_iff [IsFaithful R L M] {x y : L} :
+    toEnd R L M x = toEnd R L M y ↔ x = y :=
+  IsFaithful.injective_toEnd.eq_iff
+
+@[simp]
+lemma toEnd_eq_zero_iff [IsFaithful R L M] {x : L} :
+    toEnd R L M x = 0 ↔ x = 0 := by
+  simp [- LieHom.map_zero, ← (toEnd R L M).map_zero]
+
+lemma isFaithful_iff' : IsFaithful R L M ↔ ∀ x : L, (∀ m : M, ⁅x, m⁆ = 0) → x = 0 := by
```

Comment:
Could you please also provide an ext-lemma version. (It can't be tagged `@[ext]`, because it can't determine `M`.)
Basically `toEnd_eq_iff`, but spelled out with Lie brackets, and only the `.mpr` direction.
I guess it could be called `ext_of_isFaithful`.

Reply:
I have added `ext_of_isFaithful`.